core/src/main/java/com/datastrato/gravitino/connector/capability/HasCapabilities.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.connector.capability;
+
+public interface HasCapabilities {
+  default ColumnNotNull columnNotNull() {
+    return ColumnNotNull.supported;
+  }
+
+  default ColumnDefaultValue columnDefaultValue() {
+    throw new UnsupportedOperationException(
+        this.getClass().getName() + " does not implement columnDefaultValue capability");
+  }
+
+  default CaseSensitiveOnName caseSensitiveOnName(Capability.Scope scope) {
```

Comment:
The case-sensitive catalog name should not be managed by the capability framework but should instead be a convention of Gravitino, as catalog entities are entirely managed by Gravitino. 

Therefore, I recommend continuing to use the current case-sensitive convention for catalog names and documenting this in the user documentation.

Reply:
Of course, you can also consider supporting it in the capability module.

Reply:
I was thinking that the interface of `HasCapabilities` like this:

```java
interface HasCapabilities {

  Capability[] capabilities();
}
```

Each catalog implements this will return an array of capabilities, the framework/core uses these capabilities to apply and check. Like what @diqiu50 mentioned above rule system.

Reply:
The difference between Gravitino's capability and the rule system (such as Spark strategy) lies in the timing of the application and the target objects.

In terms of the application timing, the rule system has a single application timing, which occurs after the logical plan is generated. On the other hand, Gravitino Capability not only operates before various catalog operations dispatch but also needs to determine whether additional processing is required after the operation returns.

Regarding the target objects, the rule system acts on common Node types, allowing rules to be conveniently applied to each Node using the visitor pattern during strategy application. In contrast, Gravitino Capability needs to operate on objects like column nullable(boolean type), column default value(expression type), entity name(string type), which do not have a common parent class, thus requiring the use of generics to represent the target objects during application.